### DISCUSSION TITLE
Starting fastapi with on_event() from user-supplied CLI arguments

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [fastapi/fastapi#6055] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()
    
@app.get("/item")
def do_get_item(p3):
    return get_item()

@app.on_event("startup")
def app_startup(p1, p2):
    do_startup_stuff(p1, p2)

@app.on_event("shutdown")
def app_shutdown(p4):
    do_shutdown_stuff(p4)

def main(args):
    p1 = args[0]
    p2 = args[1]

    uvicorn.run(app)

if __name__ == "__main__":
    main(sys.argv)


Command Line 
> python myapp.py --p1 avatar2 --p2 watery
```

### Description

This question is (sort-of) related to https://github.com/tiangolo/fastapi/issues/4706#issue-1174678154.

I want to startup fastapi with user-supplied command line arguments and also initiate the on_event('startup') function. The arguments p1 and p2 differ depending on user/usage. 

The error is: `TypeError: app_startup() missing 2 required positional arguments: 'p1' and 'p2'`

How can this be achieved? Thanks

### Operating System

Windows

### FastAPI Version

fastapi-0.88.0, starlette-0.22.0

### Python Version

3.10.7

**Comment by jgould22** [comment, marked as answer]
In this case you can just do something like this.
```python

app = FastAPI()
    
@app.get("/item")
def do_get_item(p3):
    return get_item()

@app.on_event("startup")
def app_startup():
    p1 = sys.argv[0]
    p2 = sys.argv[1]
    print(p1, p2)
    # do_startup_stuff(p1, p2)

@app.on_event("shutdown")
def app_shutdown():
    p4 = sys.argv[1]
    print(p4)
    # do_shutdown_stuff(p4)


if __name__ == "__main__":
    uvicorn.run(app)

```

If these require a more complicated configuration usually in fast api this is done by using a settings/config object which you can learn more about here https://fastapi.tiangolo.com/advanced/settings/#create-the-settings-object

**Reply by dineshbvadhia** [reply]
Apologies for the tardy response. The solution is perfect for me. Thank-you so much :)
